lk2nd/lk2nd-device.c

```diff
@@ -286,6 +289,9 @@ static void lk2nd_parse_panels(const void *fdt, int offset)
 	strlcpy((char*) panel->compatible, new, new_len);
 	strlcpy((char*) panel->old_compatible, old, old_len);
 
+	if (replace_mode)
+		panel->compatible_size = new_len;
```

Comment:
we can make this even simpler, get rid of dedicated `replace_mode` variable and just write here
```c
	if (fdt_getprop(fdt, offset, "replace-compatible", NULL))
		panel->compatible_size = new_len;
```
this will make it only 2 line change to lk2nd-device.c

Reply:
This would look nicer indeed

Reply:
This didn't work nicely because `offset` variable points into wrong node inside fdt at this point in code
```
offset1:	panel {
			/* Default is Samsung S6E3FA2 */
			compatible = "samsung,s6e3fa2";
			replace-compatible;

			/* MagnaChip EA8064G */
offset2:		mdss_dsi_samsung_1080p_cmd_mag {
				compatible = "samsung,ea8064g";
			};
		};
```
offset1 is the value after `offset = fdt_subnode_offset(fdt, offset, "panel");`
offset2 is after `offset = fdt_subnode_offset(fdt, offset, panel->name);`
and `replace-compatible` is in the parent `panel` node now.

Reply:
I had to revert this attempt, I don't see how I can do this without introducing extra variables, so I guess having `bool replace_mode` is not so bad then